### PR TITLE
#3634 JSON Schema TextInput Widget

### LINKED ISSUE #3634: JSONSchemaForm TextInput widget
## Is your feature request related to a problem? Please describe.

We need a customer text input widget for rjsf

## Describe the solution you'd like

Create a new component under `JSONFormSchema/widgets/text`

## Implementation

TBD

## Describe alternatives you've considered

N/A

## Additional context

N/A

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #3634 

## Change summary

- Added a Text widget for string types
- Tried to handle focusing the next widget, seems to work OK when you use the virtual keyboard, using enter on my keyboard on the emulator has odd behaviour 🤷 
- Altered description/title fields
- The `FormTextInput` is coupled to tightly to the implementation and can't really be used for this, so just used the style and pretty much most of the same props
- The error messages are the inbuilt ones and need a bit of work to change them
- Open to style suggestions, feel free to commit

## Testing

N/A

### Related areas to think about

image

### CODE REVIEW

**Review by mark-prins — APPROVED** [review]
I like it a lot.
Yes, the style could probably be improved. It is fine as is though.

My only suggestions with style.. add some padding ( 20px on the ScrollView looks ok ) and the `SUSSOL_ORANGE` underline makes the form nicer with a bit of colour, but also looks a bit like a warning or error.
I've gone back to overall boring and used `WEARY_TARMAC` and that looks ok. but boring.

File: src/widgets/JSONForm/templates/Field.js [hunk 1]
```diff
@@ -2,22 +2,45 @@
 /* eslint-disable no-console */
```
[the remark this answers is not in the document]
Comment (joshxg) [reply src/widgets/JSONForm/templates/Field.js]: 4ea9221

File: src/widgets/JSONForm/JSONForm.js [hunk 2]
```diff
@@ -12,6 +12,7 @@ import { JSONFormWidget } from './widgets/index';
 import { JSONFormErrorList } from './JSONFormErrorList';
```
[the remark this answers is not in the document]
Comment (joshxg) [reply src/widgets/JSONForm/JSONForm.js:12]: 59a3744

**Comment by joshxg** [comment]
Thanks! I wasn't sure where to actually put it, so I left it there.. Could see it being wrapped in it's own context and having it as a component `<FocusController>` where it could be used like

```
<FocusController>
    <TextInput1/>
    <TextInput1/>
	...
    <TextInputn/>
</Focus..>
```

and focus could be handled a little easier in a few places. Also thought about overriding the base `TextInput` component and just having them register themselves somewhere and overriding the on submit to just find some other text input to focus.... why isn't it just part of RN core?! 

And sorry about the messy code - have added the prop types and removed the linter hints - oops!

**Pull request merged**